core/templates/dev/head/editor/ExplorationPreview.js

```diff
@@ -59,5 +61,31 @@ oppia.controller('ExplorationPreview', [
     $scope.$on('playerStateChange', function() {
       $scope.allParams = LearnerParamsService.getAllParams();
     });
+
+    explorationData.getData().then($scope._refreshPreviewTab = function() {
```

Comment:
I re-wrote this as you suggested, to avoid calling getData twice, and was able to get rid of refreshPreviewTab entirely.  The only complication I found was that I had to force getData not to use cached data by adding a boolean flag (otherwise the state wouldn't update properly).